packages/providers/src/transaction-request/transaction-request.ts

```diff
@@ -297,34 +297,36 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
    * @param predicate - Predicate bytes.
    * @param predicateData - Predicate data bytes.
    */
-  addMessageInput(message: MessageCoin, predicate?: AbstractPredicate) {
+  private addMessageInput(message: MessageCoinResource) {
     const { recipient, sender, amount } = message;
 
     const assetId = BaseAssetId;
 
-    let witnessIndex;
+    const input: MessageTransactionRequestInput = {
+      ...message,
+      type: InputType.Message,
+      sender: sender.toB256(),
+      recipient: recipient.toB256(),
+      amount,
+      witnessIndex: 0,
+      predicate: undefined,
+      predicateData: undefined,
+    };
 
-    if (predicate) {
-      witnessIndex = 0;
+    if ('getPredicateContent' in message) {
+      const { predicate, predicateData } = message.getPredicateContent();
+      input.predicate = predicate;
+      input.predicateData = predicateData;
     } else {
-      witnessIndex = this.getCoinInputWitnessIndexByOwner(recipient);
+      let witnessIndex = this.getCoinInputWitnessIndexByOwner(recipient);
 
       // Insert a dummy witness if no witness exists
       if (typeof witnessIndex !== 'number') {
         witnessIndex = this.createWitness();
       }
-    }
 
-    const input: MessageTransactionRequestInput = {
-      ...message,
-      type: InputType.Message,
-      sender: sender.toB256(),
-      recipient: recipient.toB256(),
-      amount,
-      witnessIndex,
-      predicate: predicate?.bytes,
-      predicateData: predicate?.predicateData,
-    };
+      input.witnessIndex = witnessIndex;
+    }
```

Comment:
If we move to the approach of making both `CoinPredicate` and `MessageCoinPredicate` be types of `Resource`, removing the `getPredicateContent` method, and given that the param `message` is being spread, I reckon there won't be a need to specifically set `predicate` and `predicateData` within the input object.

Reply:
It all depends on how we deal with [this](https://github.com/FuelLabs/fuels-ts/pull/1219#discussion_r1360678241).